src/mlx/warnings/regex_checker.py

```diff
@@ -107,12 +161,146 @@ def check(self, content):
         '''
         matches = re.finditer(self.pattern, content)
         for match in matches:
-            if (match.group('curr') == match.group('max')) and \
-                    (match.group('classification') in self.CLASSIFICATION):
-                self.count += 1
-                match_string = match.group(0).strip()
-                self.counted_warnings.append(match_string)
-                self.print_when_verbose(match_string)
+            if (classification := match.group("classification").lower()) in self.checkers:
+                self.checkers[classification].check(match)
+            else:
+                checker = CoverityClassificationChecker(classification=classification, verbose=self.verbose)
+                self.checkers[classification] = checker
+                checker.cq_enabled = self.cq_enabled
+                checker.exclude_patterns = self.exclude_patterns
+                checker.cq_description_template = self.cq_description_template
+                checker.cq_default_path = self.cq_default_path
+                checker.check(match)
+
+    def parse_config(self, config):
+        """Process configuration
+
+        Args:
+            config (dict): Content of configuration file
+        """
+        config.pop("enabled")
+        if value := config.pop("cq_description_template", None):
+            self.cq_description_template = Template(value)
+        if value := config.pop("cq_default_path", None):
+            self.cq_default_path = value
+        if value := config.pop("exclude", None):
+            self.add_patterns(value, self.exclude_patterns)
+        for classification in config:
+            classification_key = classification.lower().replace("_", " ")
+            if classification_key in CoverityClassificationChecker.SEVERITY_MAP:
+                checker = CoverityClassificationChecker(classification=classification_key, verbose=self.verbose)
+                if maximum := config[classification].get("max", 0):
+                    checker.maximum = int(maximum)
+                if minimum := config[classification].get("min", 0):
+                    checker.minimum = int(minimum)
```

Comment:
```suggestion
        for classification, checker_config in config.items():
            classification_key = classification.lower().replace("_", " ")
            if classification_key in CoverityClassificationChecker.SEVERITY_MAP:
                checker = CoverityClassificationChecker(classification=classification_key, verbose=self.verbose)
                if maximum := checker_config.get("max", 0):
                    checker.maximum = int(maximum)
                if minimum := checker_config.get("min", 0):
                    checker.minimum = int(minimum)
```
Last remark. This improves readability a bit.